imhamt/src/hamt.rs

```diff
@@ -202,3 +202,53 @@ impl<H: Default + Hasher, K: Eq + Hash, V> FromIterator<(K, V)> for Hamt<H, K, V
         h
     }
 }
+
+impl<H: Default + Hasher, K: Eq + Hash + serde::Serialize, V: serde::Serialize> serde::Serialize
+    for Hamt<H, K, V>
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use serde::ser::SerializeMap;
+        let mut map = serializer.serialize_map(Some(self.size()))?;
+        for (k, v) in self.iter() {
+            map.serialize_entry(k, v)?;
+        }
+        map.end()
+    }
+}
+
+impl<
+        'de,
+        H: Default + Hasher,
+        K: Eq + Hash + serde::Deserialize<'de>,
+        V: serde::Deserialize<'de>,
+    > serde::Deserialize<'de> for Hamt<H, K, V>
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let map = std::collections::HashMap::<K, V>::deserialize(deserializer)?;
+        Ok(Self::from_iter(map.into_iter()))
+    }
+}
+
+impl<H: Default + Hasher, K: Eq + Hash, V: PartialEq> PartialEq for Hamt<H, K, V> {
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
@vincenthz If it's guaranteed that two equal HAMTs have identical in-memory representation, we could also just use ` derive Eq` for `Hamt`.

Reply:
it's supposed to be guaranteed yes (although not really tested), although not adding the Eq was on purpose since that offers a good way to do an operation that probably shouldn't be done